src/library/scala/collection/View.scala

```diff
@@ -163,50 +163,18 @@ object View extends IterableFactory[View] {
     override def isEmpty: Boolean = underlying.isEmpty
   }
 
-  /** A class that partitions an underlying collection into two views */
-  @SerialVersionUID(3L)
-  class Partition[A](val underlying: SomeIterableOps[A], val p: A => Boolean) extends Serializable {
-
-    /** The view consisting of all elements of the underlying collection
-     *  that satisfy `p`.
-     */
-    val first = new Partitioned(this, true)
-
-    /** The view consisting of all elements of the underlying collection
-     *  that do not satisfy `p`.
-     */
-    val second = new Partitioned(this, false)
-  }
-
   /** A view representing one half of a partition. */
   @SerialVersionUID(3L)
-  class Partitioned[A](partition: Partition[A], cond: Boolean) extends AbstractView[A] {
-    def iterator = partition.underlying.iterator.filter(x => partition.p(x) == cond)
-    override def knownSize: Int = if (partition.underlying.knownSize == 0) 0 else super.knownSize
+  class Partitioned[A](val underlying: SomeIterableOps[A], val p: A => Boolean, cond: Boolean) extends AbstractView[A] {
```

Comment:
Any reason for `underlying` and `p` being public?

Reply:
@allanrenucci None I can think of, I was using the keyword `val` cause the `underlying` passed from the calling function was a `val`.  
I wasn't aware using the keyword `val` actually expose the arguments as public members and the compiler interpret the argument type as `private val` by default.
I'll remove them. :)